k2/csrc/array_ops.h

```diff
@@ -21,49 +21,12 @@
 #include "k2/csrc/context.h"
 #include "k2/csrc/log.h"
 #include "k2/csrc/ragged.h"
+#include "k2/csrc/utils.h"
 
 // Note, I'm not sure about the name of this file, they are not ops like in
 // TensorFlow, but procedures..
 
-// TODO(haowen): manage/load block config with some classes? then we can get
-// different configuration depending on num_elements and data type.
-// block size for matrix transpose.
-static constexpr int32_t kTransTileDim = 32;
-static constexpr int32_t kTransBlockRows = 8;
-
 namespace k2 {
-// TODO(haowen): move the implementations to file `op_inl.h` or
-// `op.cu`(specialized on device and data type)?
-template <typename T>
-__global__ void TransposeKernel(int32_t rows, int32_t cols, const T *input,
```

Comment:
moved to `array_ops_inl.h` and added `elementStride0` in the parameter list (my previous implementation does not support case stride0 != cols)